source/server/admin/admin.cc

```diff
@@ -615,24 +631,65 @@ ProtobufTypes::MessagePtr AdminImpl::dumpEndpointConfigs() const {
   return endpoint_config_dump;
 }
 
+ProtobufTypes::MessagePtr
+AdminImpl::dumpUnreadyTargetsConfigs(const absl::optional<std::string>& mask) const {
+  if (mask.has_value()) {
+    const std::string& mask_string = mask.value();
+    if (mask_string == "listener") {
+      return dumpListenerUnreadyTargetsConfigs();
+    }
+    // More options for unready targets config dump.
+  }
+  return std::make_unique<envoy::admin::v3::UnreadyTargetsConfigDumpList>();
+}
+
+ProtobufTypes::MessagePtr AdminImpl::dumpListenerUnreadyTargetsConfigs() const {
+  std::vector<std::reference_wrapper<Network::ListenerConfig>> listeners;
+
+  if (server_.listenerManager().isWorkerStarted()) {
+    listeners = server_.listenerManager().listeners(ListenerManager::WARMING);
+  } else {
+    listeners = server_.listenerManager().listeners(ListenerManager::ACTIVE);
+  }
+
+  auto unready_targets_config_dump_list =
+      std::make_unique<envoy::admin::v3::UnreadyTargetsConfigDumpList>();
+
+  for (auto& listener_config : listeners) {
+    auto& listener_message =
+        *unready_targets_config_dump_list->mutable_unready_targets_configs()->Add();
+    listener_message.set_name(listener_config.get().name());
+
+    auto& listener = dynamic_cast<ListenerImpl&>(listener_config.get());
+    const auto& init_manager = dynamic_cast<Init::ManagerImpl&>(listener.initManager());
```

Comment:
Can we promote the unreadyTarget() to all init::manager?
WDTY @lizan

Reply:
Great idea, will add ```unreadyTargets``` to the abstract manager as well.